core/arch/arm/kernel/generic_boot.c

```diff
@@ -309,9 +308,10 @@ static void init_runtime(unsigned long pageable_part)
 	 * Claim virtual memory which isn't paged, note that there migth be
 	 * a gap between tee_mm_vcore.lo and TEE_RAM_START which is also
 	 * claimed to avoid later allocations to get that memory.
+	 * Linear memory (flat map core memory) ends there.
 	 */
 	mm = tee_mm_alloc2(&tee_mm_vcore, tee_mm_vcore.lo,
-			(vaddr_t)(__text_init_start - tee_mm_vcore.lo));
+			(vaddr_t)(__pageable_start - tee_mm_vcore.lo));
```

Comment:
Sorry, I confused `__pageable_start` with `__text_pageable_start`.
Your change is good please keep it.